seed/php-sdk/imdb/src/Core/HttpMethod.php

```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace Seed\Core;
+
+enum HttpMethod: string
+{
+    case GET = 'GET';
+    case POST = 'POST';
+    case PUT = 'PUT';
+    case PATCH = 'PATCH';
+    case DELETE = 'DELETE';
+}
```

Comment:
any particular reason to make this string-backed rather than just the following?
```suggestion
enum HttpMethod
{
    case GET;
    case POST;
    case PUT;
    case PATCH;
    case DELETE;
}
```

Reply:
Primarily to have a simpler usage at the `buildRequest` call-site:

```php
    private function buildRequest(
        BaseApiRequest $request
    ): Request {
       $url = $this->buildUrl($request);
       $headers = $this->encodeHeaders($request);
       $body = $this->encodeRequestBody($request);
       return new Request(
           $request->method->value,
           $url,
           $headers,
           $body,
       );
    }
```

When the enum is string-backed, we can just use `$request->method->value`. Thoughts?

Reply:
i think if it's not string-backed, you can do:
```php
           $request->method->name,
```

Reply:
Oh nice, just tried that and it works well 👍